src/core/lib/iomgr/exec_ctx.cc

```diff
@@ -104,23 +107,43 @@ static void exec_ctx_sched(grpc_exec_ctx* exec_ctx, grpc_closure* closure,
   grpc_closure_list_append(&exec_ctx->closure_list, closure, error);
 }
 
-static gpr_timespec
+typedef struct time_atm_pair {
```

Comment:
Nit: this is only going to ever be used for this special case global start time. I would suggest naming it as such to be clearer. something like global_start_time_atm_pair.

Same this with the functions. 
`timespec_from_global_start_time(gpr_clock_type clock_type)`
`update_global_start_time(const gpr_timespec src)`

Reply:
I prefer to keep it since there are multiple g_start_time defined this way. Pulling definition of `g_start_time[GPR_REALTIME_CLOCK]` out of the array is feasible but I feel it makes the code less clean?

Reply:
Ah, good point, that does make more sense for the initialization loop